eth_defi/uniswap_v3/price.py

```diff
@@ -75,6 +84,481 @@ def get_amount_in(
         assert slippage >= 0
 
         encoded_path = encode_path(path, fees, exact_output=True)
-        amount_in = self.deployment.quoter.functions.quoteExactOutput(encoded_path, amount_out).call()
+        amount_in = self.deployment.quoter.functions.quoteExactOutput(
+            encoded_path, amount_out
+        ).call()
 
         return int(amount_in * (10_000 + slippage) // 10_000)
+
+
+def get_pool_fee(web3: Web3, pool_address: HexAddress):
+    """Helper function to get the swap fee of a Uniswap V3 like pool"""
+    pool_details = fetch_pool_details(web3, pool_address)
+    return pool_details.raw_fee
+
+
+def get_path_and_fees_quote_first(
+    web3: Web3,
+    base_token: HexAddress,
+    quote_token: HexAddress,
+    *,
+    intermediate_token: Optional[HexAddress] = None,
+    base_token_fee: Optional[int] = None,
+    quote_token_fee: Optional[int] = None,
+    intermediate_token_fee: Optional[int] = None,
```

Comment:
IMO the fee args for this function should be a bit better since there is no such thing as token fee